assets/src/design-system/components/input/numericInput.js

```diff
@@ -31,6 +31,7 @@ export const NumericInput = ({
   max,
   min,
   value,
+  isIndeterminate: originalIsIndeterminate,
```

Comment:
@barklund @samwhale I added this here so that if the value changes, it's not considered `isIndeterminate` anymore. So it's only indeterminate if the value has not changed and it was originally set so, too.

LMK if this makes sense to you.

Reply:
Just to confirm: this is so that the validation runs and we know that whatever value is shown is 'valid'?

Reply:
@samwhale This is related to multi-selection -- if there are multiple elements selected with different values then the value can't be determined and we should pass `isIndeterminate` and `Mixed` as the placeholder (also left you a comment [here](https://github.com/google/web-stories-wp/pull/6604#discussion_r587376838)).

Reply:
Ah okay, I'm catching up now. Thanks for the explanation 😄 

That's cool functionality that was implemented. I see why we do this now, but to be honest I don't believe the `Input` component should have the `isIndeterminate` flag added ([this addition](https://github.com/google/web-stories-wp/pull/6574/files#diff-fd6b4a2497416041553b5ae717a9a3301d1558d4b25551a583669254b2e0b4d1R140-R180)).

The `Input` is supposed to be purely presentational, so any value we want to display should be passed in. The value should be controlled from outside the input and the `Input` itself should not decide whether to display the value. The new `NumericInput` breaks that idea a bit by keeping track of 'previous' state, but I think that was necessary given the deadlines we were facing. Had we had more time (and if it wasn't such a big lift), I would have preferred to have the tracked state managed outside of the numeric input.

That being said: I know we're on a time constraint here and want to implement with the same functionality as before without going through a huge lift.

I think the best solution for the future would be to create a hook/component wrapper around the input + numeric input that decide the values to display based on the 'multiple value' changes or previous state changes. This is more of a tech debt thing, so I'd be happy to add a ticket and I can come back to it later.

What do y'all think @miina and @barklund?

Reply:
@samwhale Added this issue to follow up: https://github.com/google/web-stories-wp/issues/6640
Currently, we're using `isIndeterminate` elsewhere as well so better to address it in one place.